### PR TITLE
[SPARK-6350][Mesos] Fine-grained mode scheduler respects mesosExecutor.cores

### PR DESCRIPTION
**Pull request by dragos**
This is a regression introduced in #4960, this commit fixes it and adds a test.

@tnachen @andrewor14 please review, this should be an easy one.

### CODE REVIEW

**Comment by dragos** [comment]
I'm working on another fix for SPARK-7874, and will remove some of the duplication in tests, but I wanted to keep this PR as simple as possible (in line with the trivial fix).

File: core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala [hunk 1]
```diff
@@ -42,6 +42,38 @@ import org.apache.spark.{LocalSparkContext, SparkConf, SparkContext, SparkFunSui
...
+
+    val mesosSchedulerBackend = new MesosSchedulerBackend(taskScheduler, sc, "master")
+
+    val resources = Arrays.asList(
+      mesosSchedulerBackend.createResource("cpus", 4),
+      mesosSchedulerBackend.createResource("mem", 1024))
+    // uri is null.
+    val (executorInfo, _) = mesosSchedulerBackend.createExecutorInfo(resources, "test-id")
+    val executorResources = executorInfo.getResourcesList
+    val cpus = executorResources.asScala.find(_.getName == "cpus").get.getScalar.getValue
```
Comment (tnachen) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala]: I would prefer we use the same matching logic (equals) just to be consistent.

**Comment by tnachen** [comment]
Good catch! Indeed a regression... this PR LGTM @andrewor14

File: core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala (hunk 1 above)
Comment (dragos) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala]: You mean in the assertion below? Not sure what you mean (I think `===` offers a more detailed error message, but I am ok changing it to `==` if that's what you mean).

Comment (tnachen) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala]: I just mean _.getName.equals("cpus") :) Not a big deal, just like to be consistent

Comment (dragos) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala]: Oh, I see. I can do that. Do you know why `equals` is preferred? `==` is safe w.r.t. to nulls, so we assume `getName` (and all other getters, like `getContainerPath`) never return `null`?

Comment (tnachen) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendSuite.scala]: I don't think it's really preferred, it's just how it's written in the first place. I suggest if you think it makes sense either keep it for now or change it everywhere else.

**Comment by dragos** [comment]
Any committer wants to have a final look? It's a one-liner, but #8671 depends on this (currently I have an ignored test in there, pending this fix). /cc @andrewor14 (@srowen too, in case you have bandwidth..)

**Comment by andrewor14** [comment]
LGTM merging into master 1.5 thanks @dragos.

**Pull request closed without merging**